redisvl/extensions/router/semantic.py

```diff
@@ -241,15 +252,26 @@ def _build_aggregate_request(
             .dialect(2)
         )
 
+        filter = self._distance_threshold_filter()
+
+        aggregate_request.filter(filter)
+
         return aggregate_request
 
-    def _classify_route(
+    def _get_route_matches(
         self,
         vector: List[float],
-        distance_threshold: float,
         aggregation_method: DistanceAggregationMethod,
-    ) -> RouteMatch:
-        """Classify to a single route using a vector."""
+        max_k: int = 1,
+    ) -> List[RouteMatch]:
+        """Get the route matches for a given vector and aggregation method."""
+
+        thresholds: List[float] = [route.distance_threshold for route in self.routes]
```

Comment:
Why the variable type annotation here? Did mypy not pick up that `route.distance_threshold` is a float?